components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -82,6 +143,30 @@ WireguardKeyPair GenerateNewX25519Keypair() {
       EncodeBase64(std::vector<uint8_t>(privkey, privkey + 32)));
 }
 
+// See https://github.com/google/re2/wiki/Syntax for RE2 syntax
+bool IsValidConfig(const wchar_t* config) {
+  if (!config) {
+    return false;
+  }
+
+  std::string config_str;
+  if (!base::WideToUTF8(config, wcslen(config), &config_str)) {
+    VLOG(1) << "Unable to convert config to std::string. Is config valid?";
+    return false;
+  }
+
+  // Check for invalid characters
+  // low ascii (excluding CR/LF), #, high ascii
+  if (re2::RE2::PartialMatch(
+          config_str,
+          R"([\x00-\x09]|[\x0b-\x0c]|[\x0d-\x1f]|[\x23]|[\x7f-\xFF])")) {
```

Comment:
same as above, shouldn't we be capturing the value and using existing utilities to validate their contents?

Reply:
By the way, there is a simple INI parser in chromium under `chrome/common/ini_parser.h`, although this one wouldn´t handle multi entry for keys.

Reply:
With this, we just want to make sure none of these characters are in the config. It's not allowed